components/layout_2020/sizing.rs

```diff
@@ -100,28 +101,32 @@ pub(crate) fn outer_inline(
         .inline
         // Percentages for 'max-width' are treated as 'none'
         .and_then(|lp| lp.to_length());
-    let clamp = |l: Length| l.clamp_between_extremums(min_inline_size, max_inline_size);
+    let clamp = |l: Au| {
+        l.clamp_between_extremums(min_inline_size.into(), max_inline_size.map(|t| t.into()))
+    };
 
     let border_box_sizes = match inline_size {
         Some(non_auto) => {
-            let clamped = clamp(non_auto);
+            let clamped = clamp(non_auto.into());
             let border_box_size = match box_sizing {
-                BoxSizing::ContentBox => clamped + pb_lengths,
+                BoxSizing::ContentBox => clamped + pb_lengths.into(),
                 BoxSizing::BorderBox => clamped,
             };
             ContentSizes {
-                min_content: border_box_size,
-                max_content: border_box_size,
+                min_content: border_box_size.into(),
+                max_content: border_box_size.into(),
             }
         },
         None => get_content_size().map(|content_box_size| {
             match box_sizing {
                 // Clamp to 'min-width' and 'max-width', which are sizing the…
-                BoxSizing::ContentBox => clamp(content_box_size) + pb_lengths,
-                BoxSizing::BorderBox => clamp(content_box_size + pb_lengths),
+                BoxSizing::ContentBox => {
+                    (clamp(content_box_size.into()) + pb_lengths.into()).into()
```

Comment:
maybe you can remove extra `.into()` ?

```suggestion
                    clamp(content_box_size.into()) + pb_lengths.into()
```